apps/api/src/jobs/consumers/notifications/notification.consumer.ts

```diff
@@ -24,15 +27,55 @@ export abstract class NotificationConsumer {
     try {
       this.logger.log(`Sending notification with id: ${id}`);
       const result = await sendNotification();
-      notification.deliveryStatus = DeliveryStatus.SUCCESS;
+      notification.deliveryStatus = DeliveryStatus.AWAITING_CONFIRMATION;
+
+      if (SkipProviderConfirmationChannels.includes(notification.channelType)) {
+        notification.deliveryStatus = DeliveryStatus.SUCCESS;
+      }
+
       notification.result = { result };
     } catch (error) {
-      notification.deliveryStatus = DeliveryStatus.FAILED;
+      notification.deliveryStatus = DeliveryStatus.PENDING;
```

Comment:
This needs to be conditional, if retryCount is more than the limit we keep in in FAILED so it doesn't get picked for retry in queue itself.

Reply:
This is addressed in https://github.com/OsmosysSoftware/osmo-x/pull/263/commits/1c8fee46e6cd7bb826b8d193fbd6d1f580a3f9c7